src/network/requestQueue/index.js

```diff
@@ -300,12 +309,15 @@ module.exports = class RequestQueue extends EventEmitter {
     // will be fine, and potentially fix up a new timeout if needed at that time.
     // Note that if we're merely "overloaded" by having too many inflight requests
     // we will anyways check the queue when one of them gets fulfilled.
-    const timeUntilUnthrottled = this.throttledUntil - Date.now()
-    if (timeUntilUnthrottled > 0 && !this.throttleCheckTimeoutId) {
+    let scheduleAt = this.throttledUntil - Date.now()
+    if (!this.throttleCheckTimeoutId) {
+      if (this.pending.length > 0) {
+        scheduleAt = scheduleAt > 0 ? scheduleAt : CHECK_PENDING_REQUESTS_INTERVAL
```

Comment:
Why is it that we need a minimum timeout here? I'm thinking that if the throttle time was marginal, say 1ms, and 1ms has elapsed by the time we check it, then `this.throttledUntil - Date.now()` will be either zero or negative. At that point, couldn't we just set the timeout value to `Math.max(scheduleAt, 0)`? That would just result in the pending check request happening on the next turn around the event loop, which is what we want in this case since the throttle period has already ended and we don't want to delay the requests further.

The only problem I see with that is that we need to make sure `this.throttledUntil` is cleared after we've scheduled the check, so that we don't end up always doing `setTimeout(..., 0)` as soon as we've been throttled once.

Does that make sense?

Reply:
Thanks @Nevon for the quick review!

Doing `Math.max(scheduleAt, 0)` has the exact problem you mentioned - checkPendingRequests() and scheduleCheckPendingRequests() keep calling each other in a loop without an exit criteria. Given pending is tracking requests both because of throttling and max inflight, I felt the simplest way to implement this would be to use a minimum timeout that kicks in when pending is not empty and stop otherwise.

Reply:
@Nevon - would love to hear if you have any follow-up thoughts on the approach for this fix.

Reply:
I'm not sure I'm sold on the idea of introducing up to a 10ms delay whenever there's been marginal throttling, but it's still an improvement over the current situation. This can always be improved if it turns out that it actually happens with some amount of frequency.